### ISSUE TITLE
Make prettier pick up .editorconfig

### ISSUE THREAD

**Issue by theodm** [diffplug/spotless#1186]
Feature Request:

Prettier usually picks up the '.editorconfig'-File of the project. The following variables will be evaluated:
- end_of_line
- indent_style
- indent_size/tab_width
- max_line_length

As far as I can see the .editorconfig of the project is currently not being picked up. There's no way to configure spotless to explicitly use a .editorconfig-File. As workaround I currently reconfigure the desired properties:

```kotlin
<...>.config(prettierConfig)

val prettierConfig = mapOf("tabWidth" to 4, "printWidth" to 80)
```

So my proposal is to ideally support .editorconfig without dedicated config, as alternative the .editorconfig could be supplied as another configuration parameter.

**Comment by nedtwigg** [comment]
> the .editorconfig of the project is currently not being picked up

This is correct. If it were being accounted for, it would have to be included as part of this class

https://github.com/diffplug/spotless/blob/3ec902ccd0f7aaf441ca998971c8092d3286c578/lib/src/main/java/com/diffplug/spotless/npm/PrettierFormatterStep.java#L61-L66

so that it could be taken into consideration for up-to-date checks. Probably you would have to declare something like `prettier().editorConfig()` to trigger a search for `.editorconfig` files in this or a parent directory.

Happy to take a PR for such a feature.
